### ISSUE TITLE
Europe Improved ready to be added 

### ISSUE THREAD

**Issue by VictoryFirst2** [triplea-game/triplea#12947]
Hi, I have made improvements to the module for Axis & Allies: Europe (1999), called "Europe - Beta". Almost every single bug has been fixed and new features have been added to make as many things as possible governed by the engine.

Here is my repository: https://github.com/VictoryFirst2/europe_improved

And this is the module I improved: https://github.com/triplea-maps/europe

Perhaps it is possible that my module replaces the original one?

**Comment by beelee1** [comment]
Hi Victory

You have this as a Issue. You need to make a Pull Request to the repo you want to change. Annd then make a PR for it.

Issues are for bugs/user problems and probably other stuff :)

**Comment by VictoryFirst2** [comment]
Oh, it said in this guide I needed to create an issue xD: https://docs.google.com/document/d/1FfF7N0srp9QG0_if5D-c1d1Aa1QTttdhxgm1GBh3pI4/edit?tab=t.0

**Comment by VictoryFirst2** [comment]
So how do I create a pull request? It should be a separate module, not replace an existing module.

**Comment by beelee1** [comment]
yea so go to the repo and fork it 

then make your changes and make a PR
I think lol

**Comment by beelee1** [comment]
go to our repo and make a minor change. Change the xml by 1 number

Should be a big green buttuon show up hit that

**Comment by beelee1** [comment]
there might be access stuff going on.

You got access to EXP so try a minor one there

**Comment by beelee1** [comment]
Oh ? You want a new game basically then. Gotta make a repo from scratch and all the stuff to it then do a yaml PR

**Comment by VictoryFirst2** [comment]
Ah so I just need to edit the yaml? But shouldn't my repo be added to triplea maps?

**Comment by beelee1** [comment]
Oh yea that link is outdated

**Comment by beelee1** [comment]
hmm I struggle with this. Lets @RogerCooper if he can help

**Comment by RogerCooper** [comment]
I will try adding the game in.

**Comment by RogerCooper** [comment]
I posted into github under https://github.com/triplea-maps/europe_improved. I will add to triplea_maps.yaml tomorrow.

**Comment by VictoryFirst2** [comment]
@RogerCooper Great job! And if I would want to make a change to my module, can I just do that using Add Files? I just realized it might be a good idea to put a screenshot of the game board in the game notes.

**Comment by VictoryFirst2** [comment]
And also can the version number be 1.8? I see other maps have integers as version numbers (1, 2, 3, etc.) but mine is 1.8 😂 I hope that won't mess anything up?

**Comment by VictoryFirst2** [comment]
@RogerCooper OK so a picture of the game board in the game notes is not really necessary but I realized that the preview.png and the image in the description are still from the old map. They need to be replaced by ones of the new map but for that it says I need push access.

**Comment by VictoryFirst2** [comment]
In the new map the Middle Eastern territories are yellow instead of tan so that might cause confusion :)

**Comment by VictoryFirst2** [comment]
OK I figured it out. I created a fork but I was apparently still modifying the original module. I made changes to the images and also updated the description. It is now definitely ready for upload.

@beelee1 So I have to create a PR now?

**Comment by beelee1** [comment]
@VictoryFirst2 I would go to the repo Roger created and ask him for write access. Then you can make a test branch and add your new changes. Do the PR. Delete branch.

Then another PR is required for the map yaml. The first one takes a little setup time. After that, you just change the version number. Yaml versions must use whole numbers.

Look at existing ones. You might want Roger to do the initial one. Also the description image can be changed in the description folder and I think you might need to do something else. I forget. I still need to change EXP's

**Comment by frigoref** [comment]
@VictoryFirst2 I just skimmed through this, but it seemed that initially you had problems understanding what needs to be done when having your own map repository in order to contribute to triplea.
Could you please recheck [How to contribute to Triplea for Map Makers](https://github.com/triplea-game/triplea/blob/master/docs/contribute.md#golf-Map-Maker) and the mentioned links for what is missing.
It would be very helpful if you could you your recent experience to improve the information provided on these sites. Best would be, when you could create another PR which can be easily done per md-file by simply pressing the ![image](https://github.com/user-attachments/assets/9c806724-7d30-4296-be2f-6e2e0e8e34e7) button (edit).

@beelee1 If you have understood what was missing, feel free to contribute yourself with a PR.

**Comment by VictoryFirst2** [comment]
@frigoref Yeah I simply had some trouble navigating GitHub. I didn't know the guides you mentioned existed, I used this one: https://docs.google.com/document/d/1FfF7N0srp9QG0_if5D-c1d1Aa1QTttdhxgm1GBh3pI4/edit?tab=t.0. 

I think the guides on GitHub are fine, if I knew they were there then I think I wouldn't have had any problems.

The only thing I don't understand is that the preview.png of my map doesn't show
![image](https://github.com/user-attachments/assets/194590ab-0054-4d5e-aacb-dd74e7f3fd71)

Even though I have a preview.png in my repo. Could you help me with this?

**Comment by frigoref** [comment]
@VictoryFirst2 This is due to the wrong link in the triplea_maps.yaml file ([link](https://github.com/triplea-game/triplea/blob/caa9ca71028b3c3b1ae165021745a3841f91ffbf/triplea_maps.yaml#L2869C8-L2869C94))
![image](https://github.com/user-attachments/assets/27d07a1e-ad4b-4516-8317-7be1c5898ce0)

You might want to correct this :)

**Comment by frigoref** [comment]
> @frigoref Yeah I simply had some trouble navigating GitHub. I didn't know the guides you mentioned existed, I used this one: https://docs.google.com/document/d/1FfF7N0srp9QG0_if5D-c1d1Aa1QTttdhxgm1GBh3pI4/edit?tab=t.0.
> 
> I think the guides on GitHub are fine, if I knew they were there then I think I wouldn't have had any problems.

Do you think there is anything in this Google Docs file that is missing / better? If so, we should think to incorporate this in the triplea documentation and get rid of this separately stored documentation.

**Comment by VictoryFirst2** [comment]
@frigoref Ah, so like this? https://github.com/triplea-maps/europe_improved/blob/master/preview.png

Yeah I found the Google Doc very good, it explained how I can upload my map to GitHub with the same file structure as other maps using Git GUI. However for Step 6, it says you need to open an issue to request your map being uploaded to triplea-maps but @beelee1 said that has changed. And for Step 7, I would include some pictures that show the buttons for "fork" and "pull_request", that would guide people more to the right direction I think. 

Oh yeah if that link thingy on top is correct, then perhaps make a note of that in the documentation as all the other links have raw.githubusercontent in them and I don't understand why.

That's all, I think the guide explains everything very well. I was just afraid to do something wrong 😂

Honestly I think some video documentation about this would be very helpful. Now I understand how it works I'd be happy to make some videos someday explaining how to make maps, modify existing maps, and upload your map to triplea.

**Comment by frigoref** [comment]
@VictoryFirst2 The link is correct. 
Would be great of you could provide a PR for that.

Concerning the documentation and video: Would be great if you could provide another PR on that matter. Everything that helps is worth having it in the triplea docs folder to grow it to a good (and the only place) of place reference to go to.

**Comment by beelee1** [comment]
Hi @VictoryFirst2 

Can this be closed now ?

**Comment by VictoryFirst2** [comment]
@beelee1  Yeah frigoref asked me to create a PR for the google document but I don't know how to do that because that's on the TripleA forum. The guide for uploading maps to github.

**Comment by frigoref** [comment]
@VictoryFirst2: My idea was to move everything about the documentation into github (folder [docs](https://github.com/frigoref/triplea/tree/master/docs)). You can copy and modify the guide you have found in the TripleA forum and you can also create new md-files. The main goal is to have one place for all documentation.
@beelee1: I would like to keep this ticket open until we have also adjusted the user documentation related to it.

**Comment by VictoryFirst2** [comment]
@frigoref @beelee1 

It has been some time but I finally updated the documentation on this in Triplea docs. The guide that was there was pretty minimalistic, the one by Frostion was much better but had a bit out-of-date screenshots. I just wrote a guide that should be up-to-date and very clear for anyone who wants to upload their maps to TripleA: https://github.com/triplea-game/triplea/pull/14429

If it gets merged, this can be closed.

Maybe someday I will make a YouTube video about all of this. I already have made a video about basic TripleA map-making stuff, I wish to expand that by creating videos covering all of TripleA map-making, as well as a video dedicated to uploading maps to TripleA. Once that's done, I will reach out again so that we can discuss if it could be added to the triplea docs and precisely where.

**Comment by Cernelius** [comment]
@VictoryFirst2 A YouTube video would surely be a great help. Does @DanVanAtta or anyone have access to the official TripleA YouTube channel?
https://www.youtube.com/@triplea1241
Maybe he can give you the access to update it with new videos like the one you've already made (if you want)?

**Comment by VictoryFirst2** [comment]
@Cernelius Wow, you are flattering me. I initially planned to release my videos to my own YouTube channel, however if you value their quality so much that you agree to let them be uploaded to the official TripleA YouTube channel, that would be a great honor. I think the official TripleA YouTube really deserves to be updated with some new videos. If it contained a bunch of videos explaining how TripleA works, how to play by forum/email, how to modify/create maps, how to contribute to TripleA, that would really help new players.

I really want to contribute to TripleA in some way, but I was wondering how. If I can do something like this, that would be great!

The video I was talking about has not been uploaded to YouTube yet. It has been sitting on my computer for some time. I wanted to add a couple of small revisions before uploading. However, if my videos can get uploaded to the official TripleA YouTube channel, I will probably rerecord it to make it more professional-looking.

**Comment by VictoryFirst2** [comment]
Since Europe improved has been added and the documentation on uploading maps to TripleA has been uploaded, this issue can be closed.

@Cernelius I suspect Veqryn has access to the TripleA YouTube channel because he was the one talking in them. I am not sure if he actually gave access to the channel to other people.
